### PR TITLE
Updates for the brakeman and bundle audit

### PR DESCRIPTION
**Pull request by aprilrieger**
Updates for the brakeman and bundle audit

See audit post upgrade: https://docs.google.com/document/d/1lfwAxlsT5s2QHK0tGIAtwljVyPQSJNqQawVI65s9-c0/edit#

### CODE REVIEW

File: app/services/peaks/processor.rb [hunk 1]
```diff
@@ -27,7 +27,8 @@ def generate(remote_file)
 
       peaks
     ensure
-      system("rm -rf #{raw_path.gsub('/audio.wave', '')}") if raw_path
+      sanitized_raw_path = Shellwords.escape(raw_path.gsub('/audio.wave', ''))
```
Comment (aprilrieger) [inline app/services/peaks/processor.rb:30]: I consulted with ChatGPT for this solution. If there is a better more eloquent solution please let me know! I was having a hard time finding via google.

File: app/jobs/index_pdf_transcript_job.rb [hunk 2]
```diff
@@ -11,7 +11,9 @@ def perform(id, pdf_text)
     tmp_file = Tempfile.new
 
     tmp_file.binmode
-    `curl -o #{tmp_file.path} #{pdf_text}`
+    escaped_pdf_text = Shellwords.escape(pdf_text)
```
[the remark this answers is not in the document]
Comment (aprilrieger) [reply app/jobs/index_pdf_transcript_job.rb:14]: Confidence: Medium
Category: Command Injection
Check: Execute
Message: Possible command injection
Code: `curl -o #{Tempfile.new.path} #{pdf_text}`
File: app/jobs/index_pdf_transcript_job.rb
Line: 12

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -12,7 +12,7 @@ class ApplicationController < ActionController::Base
   def authenticate_for_staging
     if ENV['USE_HTTP_BASIC'] && !request.format.to_s.match('json') && !params[:print] && !request.path.include?('api') && !request.path.include?('pdf')
       authenticate_or_request_with_http_basic do |username, password|
-        username == "oralhistory" && password == "oralhistory"
+        username == ENV['STAGING_HTTP_BASIC_USERNAME'] && password == ENV['STAGING_HTTP_BASIC_PASSWORD']
```
Comment (aprilrieger) [inline app/controllers/application_controller.rb:15]: Brakeman Audit:
Confidence: High
Category: Basic Auth
Check: BasicAuth
Message: Basic authentication password stored in source code
Code: (password == "oralhistory")
File: app/controllers/application_controller.rb
Line: 15

File: app/services/peaks/converter.rb [hunk 4]
```diff
@@ -12,7 +12,7 @@ def fetch(remote_file)
 
     def convert(src, dst)
       dir = Dir.mktmpdir(@tmp_path, nil)
-      dst_file = "#{dir}/audio.wav"
+      dst_file = Pathname.new(dir).join('audio.wav').cleanpath.to_s
```
[the remark this answers is not in the document]
Comment (aprilrieger) [reply app/services/peaks/converter.rb:15]: Brakeman Audit:
Confidence: Medium
Category: Command Injection
Check: Execute
Message: Possible command injection
Code: spawn(sprintf("ffmpeg -loglevel panic -i %s -acodec pcm_s16le -ar 44100 %s", src, "#{Dir.mktmpdir((tmp_path or "ffmpeg-"), nil)}/audio.wav"))
File: app/services/peaks/converter.rb
Line: 19

File: app/services/peaks/processor.rb (hunk 1 above)
Comment (aprilrieger) [reply app/services/peaks/processor.rb:30]: Brakeman Audit:
Confidence: Medium
Category: Command Injection
Check: Execute
Message: Possible command injection
Code: system("rm -rf #{raw_path.gsub("/audio.wave", "")}")
File: app/services/peaks/processor.rb
Line: 29

File: app/views/interviewee/_show.html.erb [hunk 5]
```diff
@@ -7,8 +7,9 @@
   <%= react_component('MediaPlayer',
     id: parsed_children.try(:first).try(:[], "id_t").try(:gsub, '/', '-'),
     src: parsed_children.try(:first).try(:[], "url_t"),
-    peaks: parsed_children.try(:first).try(:[], "peaks"), image: "/avatar.jpg"),
-    typeOfResource: type_of_resource(document)
+    peaks: parsed_children.try(:first).try(:[], "peaks"),
```
Comment (aprilrieger) [inline app/views/interviewee/_show.html.erb:10]: Brakeman Audit:
Error: app/views/interviewee/_show.html.erb:10 :: parse error on value "," (tCOMMA) Could not parse app/views/interviewee/_show.html.erb
Location: Could not parse /Users/aprilrieger/softserv/oral-history/app/views/interviewee/_show.html.erb

File: app/views/full_text/_previous_next_doc.html.erb [hunk 6]
```diff
@@ -8,7 +8,10 @@
 
       <%= link_to_next_document @search_context[:next] %>
     </div>
+  </div>
 <% end %>
 <% if current_search_session %>
-    <div class="pull-right search-widgets">
-      <%= link_back_to_catalog class: 'btn' %>
\ No newline at end of file
+  <div class="pull-right search-widgets">
```
Comment (aprilrieger) [inline app/views/full_text/_previous_next_doc.html.erb:14]: Brakeman Audit:
Error: app/views/full_text/_previous_next_doc.html.erb:14 :: parse error on value "$" ($end) Could not parse app/views/full_text/_previous_next_doc.html.erb
Location: Could not parse Users/aprilrieger/softserv/oral-history/app/views/full_text/_previous_next_doc.html.erb

File: app/views/interviewee/_previous_next_doc.html.erb [hunk 7]
```diff
@@ -8,7 +8,10 @@
 
       <%= link_to_next_document @search_context[:next] %>
     </div>
+  </div>
 <% end %>
 <% if current_search_session %>
-    <div class="pull-right search-widgets">
-      <%= link_back_to_catalog class: 'btn' %>
\ No newline at end of file
+  <div class="pull-right search-widgets">
```
Comment (aprilrieger) [inline app/views/interviewee/_previous_next_doc.html.erb:14]: Error: app/views/interviewee/_previous_next_doc.html.erb:14 :: parse error on value "$" ($end) Could not parse app/views/interviewee/_previous_next_doc.html.erb
Location: Could not parse /Users/aprilrieger/softserv/oral-history/app/views/interviewee/_previous_next_doc.html.erb

File: .env [hunk 8]
```diff
@@ -10,6 +10,8 @@ POSTGRES_USER=postgres
 REGISTRY_HOST=index.docker.io/
 REGISTRY_URI=uclalibrary
 SITE_URI=oralhistory-test.library.ucla.edu
+STAGING_HTTP_BASIC_USERNAME=oralhistory
```
Comment (aprilrieger) [inline .env:13]: Goes with a this Brakeman Audit: https://github.com/scientist-softserv/oral-history/pull/33/files#r1144099295

File: app/controllers/application_controller.rb (hunk 3 above)
Comment (labradford) [reply app/controllers/application_controller.rb:15]: We will have to let John know that he will need to add env variables to staging and test

File: app/views/interviewee/_show.html.erb (hunk 5 above)
Comment (labradford) [reply app/views/interviewee/_show.html.erb:10]: did you see of this broke something in the UI? We are adding image and typeOfResource here for a reason.

Comment (labradford) [reply app/views/interviewee/_show.html.erb:10]: oh wait, I see it below here. sorry

**Review by labradford — APPROVED** [review]

**Pull request merged**